Add WebExt docs (#777)
parcel-bundler/website: src/recipes/web-extension.md

---
layout: layout.njk
eleventyNavigation:
  key: recipes-webext

>   title: <img src="/assets/lang-icons/webext.svg" alt=""/> Web Extension

Reviewer: Maybe add a class `dark-invert` to the stylesheet that applies `filter: invert(1)` if dark mode is enabled? Otherwise the black icon isn't visible on the dark background.

---
layout: layout.njk
eleventyNavigation:
  key: recipes-webext
  title: <img class="dark-invert" src="/assets/lang-icons/webext.svg" alt=""/> Web Extension
  order: 7
---

> Web Extension development is enabled by `@parcel/config-webext`.

Reviewer: ```suggestion
Bundling WebExtensions is enabled by using `@parcel/config-webext` like this:
```

"WebExtension" seems to be the official name for this API

> Web Extension development is enabled by `@parcel/config-webext`.

Author: ```suggestion
Bundling WebExtensions is enabled by using `@parcel/config-webextension` like this:
```

{% sample %}
{% samplefile ".parcelrc" %}

```json
{

>   "extends": "@parcel/config-webext"

Author: ```suggestion
  "extends": "@parcel/config-webextension"
```

}
```

{% endsamplefile %}
{% endsample %}

> Running Parcel on your [`manifest.json`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json) will automatically validate and bundle your web extension if you pass it as an entry.

Reviewer: ```suggestion
Running Parcel with your [`manifest.json`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json) as an entry (`parcel build manifest.json`) will automatically validate and bundle your web extension.
```

## HMR

For the best experience, you'll also want to enable a few options:

- `sourceMap#inline` and `sourceMap#inlineSources`: Source maps don't work unless inlined in web extensions
- `--host localhost`: Needed for HMR to work properly in content scripts

{% sample %}
{% samplefile "package.json" %}

```json
{
  "targets": {
    "webext-dev": {
      "sourceMap": {
        "inline": true,
        "inlineSources": true
      }
    },
    "webext-prod": {}
  },
  "scripts": {

>     "dev": "parcel src/manifest.json --host localhost --target webext-dev",

Reviewer: Please rename to `start` to be consistent with the other doc pages

    "build": "parcel build src/manifest.json --target webext-prod"
  }
}
```

{% endsamplefile %}
{% endsample %}

> Running `yarn run dev` or `npm run dev` will start the development server. HMR and source maps will work for background scripts, content scripts, the popup page, and the options page. To add the extension to your browser, research how to load an extension unpacked (for example, in Chrome, [click "Load Unpacked"](https://developer.chrome.com/extensions/getstarted#manifest)).

Reviewer: ```suggestion
Running `yarn start` or `npm start` will start the development server. HMR and source maps will work for background scripts, content scripts, the popup page, and the options page. To add the extension to your browser, research how to load an extension unpacked (for example, in Chrome, [click "Load Unpacked"](https://developer.chrome.com/extensions/getstarted#manifest)).
```
